### ISSUE TITLE
How to implement in a ReactPHP app?

### ISSUE THREAD

**Issue by acasademont** [census-instrumentation/opencensus-php#214]
Hi! Seems like the library is heavily geared towards the classic PHP single-request mode. How should I instrument my code for a ReactPHP use case? A single PHP process handles multiple requests.

Thanks!

**Comment by jcchavezs** [comment]
Would you mind to share your example code to have a clearer picture on what
you try to instrument?

Den søn. 21. okt. 2018, 21:42 skrev Albert Casademont <
notifications@github.com>:

**Comment by jcchavezs** [comment]
Is there any way to pass some sort of context downstream from the point
where we recieve the request to the implementation for example? A global
scope might not be possible as this is a concurrency model.

Den tir. 23. okt. 2018, 16:30 skrev Albert Casademont <
notifications@github.com>:

**Comment by acasademont** [comment]
@jcchavezs yes exactly. We're right now using PHP-PM which limits concurrency for the ReactPHP loops but in theory they could be running in parallel. ReactPHP has the concept of Middlwares that let's you manipulate the request object and pass some kind of context.

https://reactphp.org/http/#middleware

And in theory that ReactPHP process is never killed in the sense of a traditional php request-response, so having stuff in destructors is not recommended, it might take a long time until they are executed.

**Comment by basvanbeek** [comment]
Unfortunately the current implementation of OpenCensus Tracing expects a standard isolated PHP Request lifecycle and is thus incompatible with event driven PHP solutions like ReactPHP. 

Given enough demand this might change in the future. OpenCensus Stats should actually work quite nicely as it allows for both implicit as well as explicit scope handling.
